src/ingestion/process-event.ts

```diff
@@ -1,38 +1,455 @@
+        distinctId: string,
+        teamId: number
+    ): Promise<void> {
+        if (event === '$create_alias') {
+            await this.alias(properties['alias'], distinctId, teamId)
+        } else if (event === '$identify') {
+            if (properties['$anon_distinct_id']) {
+                await this.alias(properties['$anon_distinct_id'], distinctId, teamId)
+            }
+            // TODO: roll the two updatePersonProperties calls into one to `UPDATE posthog_person` only once
+            if (properties['$set']) {
+                this.updatePersonProperties(teamId, distinctId, properties['$set'])
+            }
+            if (properties['$set_once']) {
+                this.updatePersonProperties(teamId, distinctId, properties['$set_once'], true)
+            }
+            this.setIsIdentified(teamId, distinctId)
+        }
+    }
+
+    private async setIsIdentified(teamId: number, distinctId: string, isIdentified = true): Promise<void> {
+        let personFound = await this.db.fetchPerson(teamId, distinctId)
+        if (!personFound) {
+            try {
+                const personCreated = await this.db.createPerson(
+                    DateTime.utc(),
+                    {},
+                    teamId,
+                    null,
+                    true,
+                    new UUIDT().toString()
+                )
+                this.db.addDistinctId(personCreated, distinctId)
+            } catch {
+                // Catch race condition where in between getting and creating,
+                // another request already created this person
+                personFound = await this.db.fetchPerson(teamId, distinctId)
+            }
+        }
+        if (personFound && !personFound.is_identified) {
+            await this.db.updatePerson(personFound, { is_identified: isIdentified })
+        }
+    }
+
+    private async updatePersonProperties(
+        teamId: number,
+        distinctId: string,
+        properties: Properties,
+        setOnce = false
+    ): Promise<void> {
+        let personFound = await this.db.fetchPerson(teamId, distinctId)
+        if (!personFound) {
+            try {
+                const personCreated = await this.db.createPerson(
+                    DateTime.utc(),
+                    properties,
+                    teamId,
+                    null,
+                    false,
+                    new UUIDT().toString()
+                )
+                await this.db.addDistinctId(personCreated, distinctId)
+            } catch {
+                // Catch race condition where in between getting and creating,
+                // another request already created this person
+                personFound = await this.db.fetchPerson(teamId, distinctId)
+            }
+        }
+        this.db.updatePerson(
+            personFound!,
+            setOnce ? { ...properties, ...personFound!.properties } : { ...personFound!.properties, ...properties }
+        )
```

Comment:
This looks like a bug. It should be `updatePerson(personFound, { properties: setOnce ? {...} : {...} })`, right?

Could probably `return` or `await` this as well